packages/cli/src/lib/snapshot/vault.ts

```diff
@@ -0,0 +1,154 @@
+import PlatformClient, {
+  VaultEntryModel,
+  VaultValueType,
+  VaultVisibilityType,
+} from '@coveord/platform-client';
+import {CliUx} from '@oclif/core';
+import {yellow} from 'chalk';
+import {readJsonSync, rmSync, writeJsonSync} from 'fs-extra';
+import open from 'open';
+import {join} from 'path';
+import {cwd} from 'process';
+import dedent from 'ts-dedent';
+import {InvalidVaultEntry} from '../errors/vaultErrors';
+import {AuthenticatedClient} from '../platform/authenticatedClient';
+import {Snapshot} from './snapshot';
+import {VaultEntryAttributes} from './snapshotReporter';
+
+export class Vault {
+  private static defaultEntryValue = '';
+  public constructor(private organizationId: string) {}
+
+  public async createEntries(entries: VaultEntryAttributes[]) {
+    const client = await this.client();
+    const vaultEntryFilePath = join(cwd(), `${this.organizationId}-vault.json`);
+    this.prepareFile(vaultEntryFilePath, entries);
+
+    let valid = false;
+
+    while (!valid) {
+      await this.openFile(vaultEntryFilePath);
+      const key = await CliUx.ux.prompt('', {
+        type: 'single',
+        required: false,
+        prompt: 'Press any key once finished. Press q to abort\n',
+      });
+      if (key === 'q') {
+        return null;
+      }
```

Comment:
Using `fs.watch`, you should be able to see/know when the file is saved and thus should be able to run `this.ensureEntriesValidity` etc without having to ask the user to press a key

Reply:
mmh... I think that would be bad UX.
The user might not understand what is happening after seeing the terminal logging stuff.
Specially if the terminal and the editor are 2 separate applications. If that is the case, the user's focus will no longer be in the terminal and could miss the warning popping up. 

Anyway, we should wait for the user consent before running the next operation. It's more verbose but less confusing.

Reply:
You raise a good point about the application that would be in focus.

I think that focus hands-off should be done without additional user interactions (i.e. the user is not in charge of saying "k now the cli take back the focus", because it's just another 'useless' prompt (i.e. the user does not provide valuable info for the ongoing operation).

I think we should follow the example of Git on that regard:

![image](https://user-images.githubusercontent.com/12366410/166402887-c8db216c-6505-4652-ba87-149375258e94.png)

To ensures that the user is seeing the instructions, they put it in the doc that they open.
We should do the same, so, our file would be like this:
```json5
// Fill in your secrets, then save and close the file.
{
    "mysecret1": null,
    //...
}
```

Note that does rely on the JSON5 format `npm i JSON5`

Reply:
(Or if dotenv, just use do as git, as line starting with `#` are considered comments)